src/BookFormModal.js

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import { Form, Button, Col, Row, Container } from 'react-bootstrap';
+import reactDom from 'react-dom';
+import Modal from 'react-bootstrap/Modal'
+
+
+class BookFormModal extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      title: '',
+      description: '',
+      status: '',
+      email: ''
+    }
+  }
+
+  handleSubmit = (event) => {
+    event.preventDefault();
+    console.log(this.state)
+
+    this.props.onCreate({
+      title: this.state.title,
+      description: this.state.description,
+      status: this.state.status,
+      email: this.state.email,
+    })
+  }
+
+
+  render() {
+    // we need: onChange, onClick,
+    return (
+
+      <>
+        <Button onClick={this.props.handleShow}>
+          Add a Book
+        </Button>
+
+        <Modal show={this.props.showModal}>
+          <Modal.Header>
+            <Modal.Title>Add a Book</Modal.Title>
+          </Modal.Header>
+          <Modal.Body>
+            <Form onSubmit={this.handleSubmit}>
+              <Form.Group className="mb-3" controlId="formTitle">
+                <Form.Label>Title</Form.Label>
+                <Form.Control onChange={(event) => this.setState({ title: event.target.value })} type="name" placeholder="Enter title of book" />
+              </Form.Group>
+              <Form.Group className="mb-3" controlId="formDescription">
+                <Form.Label>Description</Form.Label>
+                <Form.Control onChange={(event) => this.setState({ description: event.target.value })} type="name" placeholder="Enter book description" />
+              </Form.Group>
+              <Form.Group className="mb-3" controlId="formStatus">
+                <Form.Label>Status</Form.Label>
+                <Form.Control onChange={(event) => this.setState({ status: event.target.value })} type="name" placeholder="Enter 'read' or 'unread'" />
+              </Form.Group>
+              <Form.Group className="mb-3" controlId="formEmail">
+                <Form.Label>Email</Form.Label>
+                <Form.Control onChange={(event) => this.setState({ email: event.target.value })} type="name" placeholder="Enter your email" />
```

Comment:
I see you guys are making progress so nice work!

Feedback here: Pull this setState call out of the Form. We should not call this so many times on the Form. Once should be sufficient and you can do this at the top of the Form. Grab a TA to help with this block 👍